src/main/kotlin/com/noumenadigital/npl/cli/commands/registry/TestCommand.kt

```diff
@@ -0,0 +1,137 @@
+package com.noumenadigital.npl.cli.commands.registry
+
+import com.noumenadigital.npl.cli.ExitCode
+import com.noumenadigital.npl.cli.exception.CommandExecutionException
+import com.noumenadigital.npl.cli.service.ColorWriter
+import com.noumenadigital.npl.cli.service.SourcesManager
+import com.noumenadigital.npl.cli.service.TestHarness
+import org.tap4j.util.StatusValues
+import java.time.Duration
+import kotlin.math.max
+
+data class TestCommand(
+    private val targetDir: String = ".",
+    private val testHarness: TestHarness = TestHarness(SourcesManager(targetDir)),
+) : CommandExecutor {
+    override val commandName: String = "test"
+    override val description: String = "Run the NPL tests"
+
+    override val parameters: List<CommandParameter> =
+        listOf(
+            CommandParameter(
+                name = "directory",
+                description = "Source directory containing NPL tests to run",
+                defaultValue = ".",
+                isRequired = false,
+            ),
+        )
+
+    override fun createInstance(params: List<String>): CommandExecutor {
+        val targetDir = params.firstOrNull() ?: parameters.find { it.name == "directory" }?.defaultValue ?: "."
+        return TestCommand(targetDir = targetDir)
+    }
+
+    override fun execute(output: ColorWriter): ExitCode {
+        try {
+            val start = System.nanoTime()
+            val testResults = testHarness.runTest()
+            val paddingResult =
+                testResults.map { it.description }.fold(25) { a, c ->
+                    max(
+                        a,
+                        c.length,
+                    )
+                }
```

Comment:
Nit, but maybe it's more readable without a fold?

```suggestion
            val paddingResult = maxOf(testResults.maxOfOrNull { it.description.length } ?: 0, 25)
```